src/sentry/integrations/slack/service.py

```diff
@@ -182,30 +191,39 @@ def notify_all_threads_for_activity(self, activity: Activity) -> None:
         slack_client = SlackSdkClient(integration_id=integration.id)
 
         # Get all parent notifications, which will have the message identifier to use to reply in a thread
-        parent_notifications = (
-            self._notification_message_repository.get_all_parent_notification_messages_by_filters(
+        with MessagingInteractionEvent(
+            interaction_type=MessagingInteractionType.GET_PARENT_NOTIFICATION,
+            spec=SlackMessagingSpec(),
+        ).capture() as lifecycle:
+            lifecycle.add_extras(
+                {
+                    "activity_id": activity.id,
+                    "group_id": activity.group.id,
+                    "project_id": activity.project.id,
+                }
+            )
+            parent_notifications = self._notification_message_repository.get_all_parent_notification_messages_by_filters(
                 group_ids=[activity.group.id],
                 project_ids=[activity.project.id],
             )
-        )
-        for parent_notification in parent_notifications:
-            try:
-                self._handle_parent_notification(
-                    parent_notification=parent_notification,
-                    notification_to_send=notification_to_send,
-                    client=slack_client,
-                )
-            except Exception as err:
-                self._logger.info(
-                    "failed to send notification",
-                    exc_info=err,
-                    extra={
-                        "activity_id": activity.id,
-                        "parent_notification_id": parent_notification.id,
-                        "notification_to_send": notification_to_send,
-                        "integration_id": integration.id,
-                    },
-                )
+            for parent_notification in parent_notifications:
+                try:
+                    self._handle_parent_notification(
+                        parent_notification=parent_notification,
+                        notification_to_send=notification_to_send,
+                        client=slack_client,
+                    )
+                except Exception as err:
```

Comment:
should this emit a failure? there are also possibly multiple parent notifications so maybe each `handle()` should = a lifecycle

Reply:
An alternative here, maybe we gather context each times this happens and then check the failure count at the end of this loop to either pass or fail it, if the ratio of failures:successes is above some threshold, we can report the entire thing as a failure instead. Having a single issue group cause multiple success/failure metrics to me seems more noisy than helpful.

Do we know why these fail?